src/main/java/com/nirmata/workflow/WorkflowManager.java

```diff
@@ -73,6 +73,16 @@
      * @return sub-Run ID
      */
     public RunId submitSubTask(RunId parentRunId, Task task);
+    
+    /**
+     * Update task progress info. This method is meant to be used inside of {@link TaskExecutor}
+     * for a task to update its execution progress.
+     *
+     * @param parentRunId run id of the parent run
+     * @param taskId the task
+     * @param current progress to be set
+     */
+    public void updateTaskProgress(RunId runId, TaskId taskId, short progress);
```

Comment:
Short is good enough for our use case as we only need decimal accuracy. Are you suggesting use float instead?